src/frontends/lean/definition_cmds.cpp

```diff
@@ -754,7 +793,7 @@ environment single_definition_cmd_core(parser & p, decl_cmd_kind kind, cmd_meta
     if (p.get_break_at_pos())
         return p.env();
 
-    bool recover_from_errors = true;
+    bool recover_from_errors = p.m_error_recovery;
```

Comment:
@kha, I changed some details of error recovery to disable it when calling the equation compiler from Lean. Can you tell me if I'm messing up?